src/services/mail.js

```diff
@@ -0,0 +1,26 @@
+/* this function is used to send emails to users
+    it takes in an object with the following properties:
+    from: the sender's email address
+    to: the recipient's email address
+    subject: the subject of the email
+    html: the html body of the email
+*/
+const nodemailer = require('nodemailer');
+
+module.exports = {
+  sendEmail: async (emailOptions) => {
+    try {
+      const transporter = nodemailer.createTransport({
+        host: 'smtp.office365.com',
+        auth: {
+          user: process.env.EMAIL_USER,
+          pass: process.env.EMAIL_PASSWORD,
+        },
+      });
+      const res = await transporter.sendMail(emailOptions);
```

Comment:
Regarding the suggestion above, I think you can add the second argument callback to fetch errors here in the `sendMail` function.

Reply:
@mehmetfatiherdem Good suggestion! Callbacks is one approach to catch errors, here @mkkasem has used a try/catch block so errors will be checked in the catch block.

Reply:
yes, we discussed it during the standup.